hudi-common/src/main/java/org/apache/hudi/common/bootstrap/index/HFileBootstrapIndex.java

```diff
@@ -200,582 +166,4 @@ public void dropIndex() {
-      try {
-        if (!closed) {
-          indexByPartitionWriter.close();
-          indexByFileIdWriter.close();
-          closed = true;
-        }
-      } catch (IOException ioe) {
-        throw new HoodieIOException(ioe.getMessage(), ioe);
-      }
-    }
-
-    @Override
-    public void begin() {
-      try {
-        HFileContext meta = new HFileContextBuilder().withCellComparator(new HoodieKVComparator()).build();
-        this.indexByPartitionWriter = HFile.getWriterFactory(metaClient.getStorageConf().unwrapAs(Configuration.class),
-                new CacheConfig(metaClient.getStorageConf().unwrapAs(Configuration.class)))
-            .withPath((FileSystem) metaClient.getStorage().getFileSystem(), new Path(indexByPartitionPath.toUri()))
-            .withFileContext(meta).create();
-        this.indexByFileIdWriter = HFile.getWriterFactory(metaClient.getStorageConf().unwrapAs(Configuration.class),
-                new CacheConfig(metaClient.getStorageConf().unwrapAs(Configuration.class)))
-            .withPath((FileSystem) metaClient.getStorage().getFileSystem(), new Path(indexByFileIdPath.toUri()))
-            .withFileContext(meta).create();
-      } catch (IOException ioe) {
-        throw new HoodieIOException(ioe.getMessage(), ioe);
-      }
-    }
-
-    @Override
-    public void appendNextPartition(String partitionPath, List<BootstrapFileMapping> bootstrapFileMappings) {
-      sourceFileMappings.put(partitionPath, bootstrapFileMappings);
-    }
-
-    @Override
-    public void finish() {
-      // Sort and write
-      List<String> partitions = sourceFileMappings.keySet().stream().sorted().collect(Collectors.toList());
-      partitions.forEach(p -> writeNextPartition(p, sourceFileMappings.get(p).get(0).getBootstrapPartitionPath(),
-          sourceFileMappings.get(p)));
-      sourceFileMappings.values().stream().flatMap(Collection::stream).sorted()
-          .forEach(this::writeNextSourceFileMapping);
-      commit();
-    }
-  }
-
-  /**
-   * IMPORTANT :
-   * HFile Readers use HFile name (instead of path) as cache key. This could be fine as long
-   * as file names are UUIDs. For bootstrap, we are using well-known index names.
-   * Hence, this hacky workaround to return full path string from Path subclass and pass it to reader.
-   * The other option is to disable block cache for Bootstrap which again involves some custom code
-   * as there is no API to disable cache.
-   */
-  private static class HFilePathForReader extends Path {
-
-    public HFilePathForReader(String pathString) throws IllegalArgumentException {
-      super(pathString);
-    }
-
-    @Override
-    public String getName() {
-      return toString();
-    }
-  }
-
-  /**
-   * This class is explicitly used as Key Comparator to workaround hard coded
-   * legacy format class names inside HBase. Otherwise we will face issues with shading.
-   */
-  public static class HoodieKVComparator extends CellComparatorImpl {
```

Comment:
The fully-qualified class name of the cell comparator is written to the [HFile Trailer](https://github.com/apache/hudi/blob/master/hudi-io/hfile_format.md#trailer).  The class change can lead to backward incompatibility, i.e., HBase HFile reader may not be able to read the old HFiles referencing this class.  Also check #5004 for more details.  We need to test this and see if that's an issue.  Note that the our native HFile reader does not use the cell comparator name from the HFile trailer so there is no compatibility issue with the native HFile reader.